src/content/en/fundamentals/media/manipulating/live-effects.md

```diff
@@ -0,0 +1,374 @@
+the right and sum them all together with four times the value of the current pixel. Then you divide
+the result by 8. You can see how the result will be a pixel that is mostly the original, but with
+the nearby pixels bleeding in.
+
+    const kernel = [[0, 1, 0],
+                    [1, 4, 1],
+                    [0, 1, 0]];
+
+    for (let y = 0; y < imageHeight; y++) {
+      for (let x = 0; x < imageWidth; x++) {
+        let redTotal = 0;
+        let greenTotal = 0;
+        let blueTotal = 0;
+        let weightTotal = 0;
+        for (let i = -1; i <= 1; i++) {
+          for (let j = -1; j <= 1; j++) {
+            // Filter out pixels that are off the edge of the image
+            if (x + i > 0 && x + i < imageWidth &&
+                y + j > 0 && y + j < imageHeight) {
+              const index = (x + i + (y + j) * imageWidth) * 4;
+              const weight = kernel[i + 1][j + 1];
+              redTotal += weight * originalPixels[index];
+              greenTotal += weight * originalPixels[index + 1];
+              blueTotal += weight * originalPixels[index + 2];
+              weightTotal += weight;
+            }
+          }
+        }
+
+        const outputIndex = (x + y * imageWidth) * 4;
+        imageData.data[outputIndex] = redTotal / weightTotal;
+        imageData.data[outputIndex + 1] = greenTotal / weightTotal;
+        imageData.data[outputIndex + 2] = blueTotal / weightTotal;
+      }
+    }
+
+This gives the basic idea, but there are many guides out there that will go into much more
+detail, and list many other useful kernels.
+
+### Whole image
+Some whole image transformations are simple. In a 2D canvas, cropping and scaling is a simple case
+of only drawing part of the source image onto the canvas.
+
+    // Set the canvas to be a little smaller than the original image
+    canvas.width = source.naturalWidth - 100;
+    canvas.height = source.naturalHeight - 100;
+
+    // Draw only part of the image onto the canvas
+    const sx = 50; // The left-most part of the source image to copy
+    const sy = 50; // The top-most part of the source image to copy
+    const sw = source.naturalWidth - 100; // How wide the rectangle to copy is
+    const sh = source.naturalHeight - 100; // How tall the rectangle to copy is
+
+    const dx = 0; // The left-most part of the canvas to draw over
+    const dy = 0; // The top-most part of the canvas to draw over
+    const dw = canvas.width; // How wide the rectangle to draw over is
+    const dh = canvas.height; // How tall the rectangle to draw over is
+
+    context.drawImage(theOriginalImage,
+        sx, sy, sw, sh,
+        dx, dy, dw, dh);
+
+Rotation and reflection are directly available in the 2D context. Before you draw the image into the
+canvas, change the various transforms.
+
+    // Move the canvas so that the center of the canvas is on the Y-axis
+    context.translate(-canvas.width / 2, 0);
+
+    // An X scale of -1 reflects the canvas in the Y-axis
```

Comment:
Your call, really. I just found it ambiguous: The word “reflects” could be seen to mean “convey”.